gm4_block_compressors/data/gm4_block_compressors/functions/item/compress.mcfunction

```diff
@@ -0,0 +1,22 @@
+# @s = item entity to be compressed
+# at location of dropper (compressor)
+# run from block_compressors:item
+# compresses item by setting count, storing current item tag, and adding lore/enchantment glint
+
+# copy existing data for restoration at decompression, second line fixes bug where gm4_block_compressors gets saved to old_tag
+data modify storage gm4_block_compressors:temp/item_stack Item.tag.gm4_block_compressors.old_tag set from storage gm4_block_compressors:temp/item_stack Item.tag
+data remove storage gm4_block_compressors:temp/item_stack Item.tag.gm4_block_compressors.old_tag.gm4_block_compressors
+
+# add tag to indicate store compression size
+execute store result storage gm4_block_compressors:temp/item_stack Item.tag.gm4_block_compressors.compression_level int 1 run data get storage gm4_block_compressors:temp/item_stack Item.Count
```

Comment:
I believe you can use an item modifier to replace most of this function. (`set_count` with 1, `set_lore` using the score, `copy_nbt` for old_tag and for `compression_level`, `set_nbt` for enchantment glint)

Reply:
Talked to misode, said not to do this because one can't use item modifiers on items entities, and I don't know if copying to the armor stand's hand and then using a modifier would really be an improvement.